python/lang/best-practice/unspecified-open-encoding.yaml

```diff
@@ -0,0 +1,36 @@
+rules:
+- id: unspecified-open-encoding
+  patterns:
+  - pattern-inside: open(...)
+  - pattern-not: open(..., encoding="...", ...)
+  - pattern-not: open($F, "...", $B, "...", ...)
+  - pattern-either:
+    - pattern: open($FILE)
+    - patterns:
+      - pattern: open($FILE, ...)
+      - pattern-not: open($FILE, $M, ...)
+      - pattern-not-regex: open\(.*(?:encoding|mode)=.*\)
+    - patterns:
+      - pattern: open($FILE, $MODE, ...)
+      - metavariable-regex:
+          metavariable: $MODE
+          regex: (?!.*b.*)
+    - patterns:
+      - pattern: open($FILE, ..., mode=$MODE, ...)
+      - metavariable-regex:
+          metavariable: $MODE
+          regex: (?!.*b.*)
+
+  message: >-
+    Missing 'encoding' parameter.
+    'open()' uses device locale encodings by default, corrupting files with special characters.
+    Specify the encoding to ensure cross-platform support when opening files in text mode (e.g. encoding="utf-8").
+  languages: [python]
+  severity: ERROR
```

Comment:
nit: Mark this as a WARNING since it won't immediately crash the program and since the locale encoding might be right some of the time

Reply:
You're right - for some reason I thought `encoding="locale"` from [here](https://www.python.org/dev/peps/pep-0597/#encoding-locale) was implemented, but it was just a proposal. Corrected this in the recent commit.